ocaml/xapi/storage_smapiv1_migrate.ml

```diff
@@ -569,6 +569,14 @@ module MIRROR : SMAPIv2_MIRROR = struct
       Storage_migrate_helper.get_remote_backend url verify_dest
     in
     match remote_mirror with
+    | Mirror.QCOW2_mirror _ ->
+        (* this should never happen *)
```

Comment:
This may not be the case if @gthvn1 gets away with it :)

Reply:
The type of the mirror is based on the SMAPI version reported by each storage backend (although they are currently hardcoded). So I don't think this will be an issue unless one changes a SMAPIv1 driver to a SMAPIv3 driver. Feel free to change this bit in the future if needed though